### DISCUSSION TITLE
Moving from mow.cli to cobra?

### DISCUSSION THREAD

**Discussion by b00f** [pactus-project/pactus#198] [Q&A] [answered]
Currently we are using [mow.cli](https://github.com/jawher/mow.cli) package for CLI.
Any one has experience with [cobra](https://github.com/spf13/cobra)? 
What is the good points if we move to Cobra?

**Comment by Ali-Meh** [comment]
as of my own experience they are much like each other just cobra won game and has bigger community and active release pipe besides it has OS specific customization like autocomplete and stuff other than community I'd say they are basically same just cobra codes are more reusable even though never occurred to myself if we migrate to cobra we might to opt in [promptui](https://github.com/manifoldco/promptui) as well

**Reply by b00f** [reply]
Does Cobra fully support auto-completion feature?

**Comment by yenkhoon** [comment, marked as answer]
are we switching to cobra?

**Reply by b00f** [reply]
Yes, If we can, because it provides some tools like auto-completion

**Comment by ja7ad** [comment]
@b00f time now is for refactoring cli package to the cobra.

**Reply by b00f** [reply]
Yes, This is a long TODO. @yenkhoon wanted to work on it. I don't know if he had any progress on that

**Reply by ja7ad** [reply]
We can make an issue for the first label to replace the cobra with cli.

**Reply by b00f** [reply]
We have: https://github.com/pactus-project/pactus/issues/288

**Reply by ja7ad** [reply]
> We have: https://github.com/pactus-project/pactus/issues/288
> 

I think this issue needs to be related to the new issue.
